src/sqlitestorage.cpp

```diff
@@ -1848,7 +1848,19 @@ void SqliteStorage::calendarModified(bool modified, Calendar *calendar)
 
 void SqliteStorage::calendarIncidenceAdded(const Incidence::Ptr &incidence)
 {
-    if (!d->mIncidencesToInsert.contains(incidence->uid(), incidence) && !d->mIsLoading) {
+    QMultiHash<QString, KCalendarCore::Incidence::Ptr>::Iterator deleted =
+        d->mIncidencesToDelete.find(incidence->uid());
+    if (deleted != d->mIncidencesToDelete.end() && !d->mIsLoading) {
+        qCDebug(lcMkcal) << "removing incidence from deleted" << incidence->uid();
+        while (deleted != d->mIncidencesToDelete.end()) {
+            if ((*deleted)->recurrenceId() == incidence->recurrenceId()) {
+                deleted = d->mIncidencesToDelete.erase(deleted);
+                calendarIncidenceChanged(incidence);
```

Comment:
I assume this method updates the incidence in the database with the new values?

Reply:
Exact, it's the one that is called when an incidence is modified.